### PR TITLE
Using counter

### PR DESCRIPTION
**Pull request by Guy-Markman**

### CODE REVIEW

**Comment by Guy-Markman** [comment]
@alonbl

File: EX04/08-daemon.py [hunk 1]
```diff
@@ -4,12 +4,31 @@
...
+counter = 0
 
 
 def set_ign(signum, frame):
     signal.signal(signum, signal.SIG_IGN)
 
 
+def terminate_handler(signum, frame):
+    global counter
+    counter = -100000
```
Comment (alonbl) [inline EX04/08-daemon.py]: why use magics? please be explicit.

File: EX04/08-daemon.py [hunk 2]
```diff
@@ -4,12 +4,31 @@
...
+
+
+def add_handler(signum, frame):
+    global counter
+    counter += 1
+
+
 def set_up():
     os.closerange(3, resource.RLIMIT_NOFILE)
     new_directory = os.open("/dev/null", os.O_WRONLY)
```
Comment (alonbl) [inline EX04/08-daemon.py]: what's wrong here?

File: EX04/08-daemon.py [hunk 3]
```diff
@@ -4,12 +4,31 @@
...
+    counter = 0
+
+
+def add_handler(signum, frame):
+    global counter
+    counter += 1
+
+
 def set_up():
     os.closerange(3, resource.RLIMIT_NOFILE)
```
Comment (alonbl) [inline EX04/08-daemon.py]: please read what RLIMIT_NOFILE is and how to use it

File: EX04/08-daemon.py [hunk 4]
```diff
@@ -19,31 +38,25 @@ def set_up():
...
-    signal.signal(signal.SIGALRM, set_ign)
+    signal.signal(signal.SIGUSR1, reset_handler)
+    signal.signal(signal.SIGALRM, add_handler)
+    signal.alarm(1)
 
 
 def proc_child():
-    counter = 0
+    print counter
     log_fd = os.open("log_file.txt", os.O_CREAT | os.O_APPEND, 00777)
```
Comment (alonbl) [inline EX04/08-daemon.py]: mode is wrong

File: EX04/08-daemon.py [hunk 5]
```diff
@@ -19,31 +38,25 @@ def set_up():
...
-    signal.signal(signal.SIGINT, set_ign)
-    signal.signal(signal.SIGTERM, set_ign)
+    signal.signal(signal.SIGINT, terminate_handler)
+    signal.signal(signal.SIGTERM, terminate_handler)
     signal.signal(signal.SIGHUP, signal.SIG_IGN)
-    signal.signal(signal.SIGUSR1, set_ign)
-    signal.signal(signal.SIGALRM, set_ign)
+    signal.signal(signal.SIGUSR1, reset_handler)
+    signal.signal(signal.SIGALRM, add_handler)
+    signal.alarm(1)
```
Comment (alonbl) [inline EX04/08-daemon.py]: this should not be part of setup

File: EX04/08-daemon.py [hunk 6]
```diff
@@ -4,12 +4,31 @@
...
 import signal
 import util
+import time
+
+
+counter = 0
 
 
 def set_ign(signum, frame):
     signal.signal(signum, signal.SIG_IGN)
```
Comment (alonbl) [inline EX04/08-daemon.py]: why do you need this?

Comment (Guy-Markman) [inline EX04/08-daemon.py]: Forgot to delete

File: EX04/08-daemon.py (hunk 1 above)
Comment (Guy-Markman) [inline EX04/08-daemon.py]: changed

File: EX04/08-daemon.py (hunk 3 above)
Comment (Guy-Markman) [inline EX04/08-daemon.py]: changed

File: EX04/08-daemon.py (hunk 5 above)
Comment (Guy-Markman) [inline EX04/08-daemon.py]: Moved to proc_child

File: EX04/08-daemon.py (hunk 2 above)
Comment (Guy-Markman) [inline EX04/08-daemon.py]: Changed to RDWR.

File: EX04/08-daemon.py (hunk 4 above)
Comment (Guy-Markman) [inline EX04/08-daemon.py]: I think the mode should be: 
User - read\ write
Group - read (so other people could help too) but no to change anything
Other - read (for the same reason of Group)

So I think the mode should be 0o644.
Am I right?

**Comment by Guy-Markman** [comment]
I think it should be perfect now

**Comment by Guy-Markman** [comment]
@alonbl 
Do you have any comments? May I merge and continue to work only on the Proxy Server

**Pull request merged**